src/BIAnalyticsService/RetrieveData/RetrieveData.cs

```diff
@@ -0,0 +1,122 @@
+using System.Net;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.Functions.Worker.Http;
+using Microsoft.Extensions.Logging;
+using System.Text.Json;
+using NHS.ServiceInsights.Common;
+using NHS.ServiceInsights.Model;
+
+namespace NHS.ServiceInsights.BIAnalyticsService;
+
+public class RetrieveData
+{
+    private readonly ILogger<RetrieveData> _logger;
+    private readonly IHttpRequestService _httpRequestService;
+    public RetrieveData(ILogger<RetrieveData> logger, IHttpRequestService httpRequestService)
+    {
+        _logger = logger;
+        _httpRequestService = httpRequestService;
+    }
+
+    [Function("RetrieveData")]
+    public async Task<HttpResponseData> Run([HttpTrigger(AuthorizationLevel.Anonymous, "get")] HttpRequestData req)
+    {
+        _logger.LogInformation("Request to retrieve episode information has been processed.");
+
+        string episodeId = req.Query["EpisodeId"];
+
+        if (string.IsNullOrEmpty(episodeId))
+        {
+            _logger.LogError("Please enter a valid Episode ID.");
+            var badRequestResponse = req.CreateResponse(HttpStatusCode.BadRequest);
+            return badRequestResponse;
```

Comment:
Can just do this on one line e.g. `return req.CreateResponse(HttpStatusCode.BadRequest);`

Same for the other return statements